File: tests/modules/iter.rs
use deno_bindgen3::js;
use deno_runtime::deno_core;

#[js(module(import("./iter.js"), fast, side_module))]
pub struct Iter;

#[js(properties)]
impl Iter {
    #[js(func)]
    pub fn fibonacci(&self, iter: usize) -> v8<Fibonacci> {}
}

#[js(value)]
struct Fibonacci;

#[js(iterator)]
impl Fibonacci {
    type Item = usize;
}

File: crates/ferrosaur/examples/js/iter.js
/**
 * @param {number} iter
 */
export function* fibonacci(iter) {
  let a = 0;
  let b = 1;
  for (let i = 0; i < iter; i++) {
    yield a;
    [a, b] = [b, a + b];
  }
}
